fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/common/DataDefinitionUtil.java

```diff
@@ -34,59 +34,61 @@
      */
     public static String createUniqueIndex(String schemaName, String tableName, String indexName, List<String> indexColumns,
             List<String> includeColumns) {
-        
+
         StringBuilder result = new StringBuilder();
         result.append(createUniqueIndex(schemaName, tableName, indexName, indexColumns));
         result.append(" INCLUDE (");
         result.append(join(includeColumns));
         result.append(")");
-        
+
         return result.toString();
     }
 
     /**
-     * 
+     *
      * @param schemaName
      * @param tableName
      * @param indexName
      * @param indexColumns
      * @return
      */
     public static String createUniqueIndex(String schemaName, String tableName, String indexName, List<String> indexColumns) {
-        
+
         StringBuilder result = new StringBuilder();
         result.append("CREATE UNIQUE INDEX ");
-        result.append(getQualifiedName(schemaName, indexName));
+        // Postgresql doesn't support index name prefixed with the schema name.
+        result.append(indexName);
         result.append(" ON ");
         result.append(getQualifiedName(schemaName, tableName));
         result.append("(");
         result.append(join(indexColumns));
         result.append(")");
-                
+
         return result.toString();
     }
 
 
     /**
      * Create the DDL for a plain old index
-     * 
+     *
      * @param schemaName
      * @param tableName
      * @param indexName
      * @param indexColumns
      * @return
      */
     public static String createIndex(String schemaName, String tableName, String indexName, List<String> indexColumns) {
-                
+
         StringBuilder result = new StringBuilder();
         result.append("CREATE INDEX ");
-        result.append(getQualifiedName(schemaName, indexName));
```

Comment:
This is good defensive programming. it's actually checking against a whitelist, and prevents any injection, since this is bundled/shipped, it's worth keeping, and adding a postgres branching logic.

Reply:
good idea. added logic to do this for postgresql only.